### ISSUE TITLE
Upgrade scala.js to 1.0.1

### ISSUE THREAD

**Issue by aparo** [7mind/izumi#952]
The actual version of scala.js is 0.6.32. 
It's better to support the latest version of scala.js because it's blocking the update of zio library due to its dependency on zio/izumi-reflect.
Also logstage cannot be used on scala.js 1.0.0 projects.

**Comment by neko-kai** [comment]
@aparo Latest `izumi-reflect` is already on Scala.js `1.0`, so it's not blocking ZIO, but currently this main Izumi repo is actually blocked on ZIO itself - ZIO decided to postpone upgrade till ZIO-1.0 release, see this PR's latest comments for details - https://github.com/zio/zio/pull/2479

The best thing you can do is contribute CI scripts for cross-publishing 0.6 & 1.0 to ZIO - then they'll be able to release both versions and we in turn will be able to switch to 1.0.
